### PR TITLE
Add a variant to publish libs to Bintray

### PR DESCRIPTION
**Pull request by ennru**
* Lock the jabba version
* Add `PlayLibraryBintrayBase` as alternative to `PlayLibraryBase`
* Add some sbt settings that end up in the pom.xml

### CODE REVIEW

**Comment by ennru** [comment]
I'm not really sure about what I'm doing... seems to be a great place to go wild.

**Comment by ennru** [comment]
I tried this setup on play-mailer.

File: src/main/scala/interplay/PlaySonatypeBase.scala [hunk 1]
```diff
@@ -13,6 +14,7 @@ object PlaySonatypeBase extends AutoPlugin {
   override def requires = Sonatype
 
   override def projectSettings = Seq(
+    playBuildPromoteSonatype in ThisBuild := true,
```
Comment (ignasi35) [inline src/main/scala/interplay/PlaySonatypeBase.scala:17]: Doesn't this interfere with https://github.com/playframework/interplay/blob/master/src/main/scala/interplay/PlayReleaseBase.scala#L51-L54 ?

Comment (ennru) [reply src/main/scala/interplay/PlaySonatypeBase.scala:17]: Yes, this is how the Sonatype releasing selected.

Comment (ignasi35) [reply src/main/scala/interplay/PlaySonatypeBase.scala:17]: I think adding that is causing the error in the scripted test:

```
[info] [error] [MISSING_CREDENTIAL] No credential is found for oss.sonatype.org. Prepare ~/.sbt/(sbt_version)/sonatype.sbt file.
```

Comment (ignasi35) [reply src/main/scala/interplay/PlaySonatypeBase.scala:17]: I can no longer reproduce. 

I don't know what's going on.

File: src/main/scala/interplay/PlayLibraryBase.scala [hunk 2]
```diff
@@ -12,15 +12,14 @@ import interplay.Omnidoc.autoImport.omnidocTagPrefix
...
+object PlayLibraryBase extends AutoPlugin {
 
   override def trigger = noTrigger
-  override def requires = PlayBuildBase && PlaySonatypeBase && Omnidoc
+  override def requires = PlayBuildBase && Omnidoc
 
   import PlayBuildBase.autoImport._
 
   override def projectSettings = Seq(
-    playBuildPromoteSonatype in ThisBuild := true,
```
Comment (ignasi35) [inline src/main/scala/interplay/PlayLibraryBase.scala:23]: 🤦🏼‍♂️ that was moved from here!

File: src/main/scala/interplay/PlaySonatypeBase.scala (hunk 1 above)
Comment (ignasi35) [inline src/main/scala/interplay/PlaySonatypeBase.scala:17]: Adding this here impacts `PlayRootProjectBase` which, for some reason, `requires` `PlaySonatypeBase` even though it's a [`configuration for a Play Root Project that doesn't get published`](https://github.com/playframework/interplay/blob/d058d2c3671dc6ce6d352b6ef50b9b739c9101d2/src/main/scala/interplay/PlayBuildBase.scala#L80-L83). 

As a consequence, the `release` step on the `scripted` tests fails.

I think the correct solution is to apply the change:

```diff
-  override def requires = PlayBuildBase && PlayBintrayBase && PlaySonatypeBase && PlayReleaseBase
+  override def requires = PlayBuildBase && PlayReleaseBase
```

YOLO

**Comment by ennru** [comment]
Sorry, saw your commit too late...

**Comment by ignasi35** [comment]
> Sorry, saw your commit too late...

No worries. We seem to agree on the cause (location of the release target flags).

Still, I think the patch I suggested in https://github.com/playframework/interplay/pull/118#discussion_r426871038 still applies.

**Comment by ignasi35** [comment]
This LGTM now. 

I think we should duplicate at least one of the `scripted` tests to assert the published artifacts will properly go to bintray (e.g. https://github.com/playframework/interplay/blob/bc50d54deff4e1cfd457950146b01d61747698e2/src/sbt-test/interplay/library-with-plugin/test#L24-L26)

**Comment by ennru** [comment]
I created a scripted test for `PlayLibraryToBintray` now.

`library-to-bintray-with-plugin` checks
```
> contains mock-library/target/scala-2.12/publish-version Bintray-Maven-Publish-playframework-maven-mock:1.2.3
> contains mock-library/target/scala-2.13/publish-version Bintray-Maven-Publish-playframework-maven-mock:1.2.3
```

instead of `library-with-plugin`'s
```
> contains mock-library/target/scala-2.12/publish-version sonatype-local-bundle:1.2.3
> contains mock-library/target/scala-2.13/publish-version sonatype-local-bundle:1.2.3
```

**Comment by ennru** [comment]
We'll remove Interplay from most repositories instead of messing more with it.

**Pull request closed without merging**